lib/modules/datasource/rpm/index.ts

```diff
@@ -0,0 +1,220 @@
+import { gunzip } from 'node:zlib';
+import { promisify } from 'util';
+import { XmlDocument } from 'xmldoc';
+import { logger } from '../../../logger';
+import { cache } from '../../../util/cache/package/decorator';
+import { joinUrlParts } from '../../../util/url';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+const gunzipAsync = promisify(gunzip);
+export class RpmDatasource extends Datasource {
+  static readonly id = 'rpm';
+
+  // repomd.xml is a standard file name in RPM repositories which contains metadata about the repository
+  static readonly repomdXmlFileName = 'repomd.xml';
+
+  constructor() {
+    super(RpmDatasource.id);
+  }
+
+  /**
+   * Users are able to specify custom RPM repositories as long as they follow the format.
+   * There is a URI http://linux.duke.edu/metadata/filelists in the <sha>-filelists.xml.
+   * But according to this post, it's not something we can really look into or reference.
+   * @see{https://lists.rpm.org/pipermail/rpm-ecosystem/2015-October/000283.html}
+   */
+  override readonly customRegistrySupport = true;
+
+  /**
+   * Fetches the release information for a given package from the registry URL.
+   *
+   * @param registryUrl - the registryUrl should be the folder which contains repodata.xml and its corresponding file list <sha256>-filelists.xml.gz, e.g.: https://packages.microsoft.com/azurelinux/3.0/prod/cloud-native/x86_64/repodata/
+   * @param packageName - the name of the package to fetch releases for.
+   * @returns The release result if the package is found, otherwise null.
+   */
+  @cache({
+    namespace: `datasource-${RpmDatasource.id}`,
+    key: ({ registryUrl, packageName }: GetReleasesConfig) => packageName,
+    ttlMinutes: 1440,
+  })
+  async getReleases({
+    registryUrl,
+    packageName,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    if (!registryUrl || !packageName) {
+      return null;
+    }
+    try {
+      const filelistsGzipUrl = await this.getFilelistsGzipUrl(registryUrl);
+      if (!filelistsGzipUrl) {
+        return null;
+      }
+      return await this.getReleasesByPackageName(filelistsGzipUrl, packageName);
+    } catch (err) {
+      this.handleGenericErrors(err);
+    }
+  }
+
+  // Fetches the filelists.xml.gz URL from the repomd.xml file.
+  @cache({
+    namespace: `datasource-${RpmDatasource.id}`,
+    key: ({ registryUrl }: GetReleasesConfig) =>
+      registryUrl ?? '__no_registry_url__',
```

Comment:
```suggestion
    key: (registryUrl: string) => registryUrl,
```

must match the function arguments below